operator/v1/types_network.go

```diff
@@ -433,6 +447,19 @@ type OVNKubernetesConfig struct {
 	// fields within ipv4 for details of default values.
 	// +optional
 	IPv6 *IPv6OVNKubernetesConfig `json:"ipv6,omitempty"`
+
+	// routeAdvertisements determines if the functionality to advertise cluster
+	// network routes through a dynamic routing protocol, such as BGP, is
+	// enabled or not. This functionality is configured through the
+	// ovn-kubernetes RouteAdvertisements CRD. Requires the 'FRRK8s' routing
+	// capability provider to be enabled as an additional routing capability.
+	// Allowed values are "Enabled", "Disabled" and ommited. When omitted, this
+	// means the user has no opinion and the platform is left to choose
+	// reasonable defaults. These defaults are subject to change over time. The
+	// current default is "Disabled".
+	// +openshift:enable:FeatureGate=RouteAdvertisements
+	// +optional
+	RouteAdvertisements RouteAdvertisementsEnablement `json:"routeAdvertisements,omitempty"`
```

Comment:
Just for my understanding, does this mean that the RouteAdvertisements CRD is not installed unless this is enabled?

Reply:
If this is not enabled, the corresponding controller(s) in OVN-K won't run. I guess the plan is to not install the CRD unless this is enabled, yeah. Would that be a factor for this API?

Reply:
If you then disable this, I would expect your controller then to delete the CRD right? In which case, you might lose data? Or something? 

I'm just wondering if you should be allowed to remove this feature, if you can remove this feature, how does that impact the cluster?

Reply:
I didn't think on the perspective of data loss. So what is recommended here? Have the CRD in the system regardless of whether it is enabled or not so that we preserve instances even if disabled so that the config is not lost?

Reply:
Yes, I think I would potentially go for, once it's installed, you don't remove it, even if the capability is no longer enabled. That was it is left to the user to delete the CRD if they truly want to

You'll probably want to document the consequences of disabling the capability, and explain that the CRD will be left around and users can remove it if they want to

In other cases where we have optional capabilities like this, I think we avoided this issue by not allowing a capability to be disabled once it was enabled